portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/handlers/FavoritesHandler.java

```diff
@@ -134,10 +138,10 @@ public UserFavorites userFavorites(String userId)
 
 		result.autoFavoritesEnabled = autoFavorite;
 
-		List<String> favoriteSiteIds = (List<String>)props.getPropertyList(FAVORITES_PROPERTY);
-
-		if (favoriteSiteIds == null) {
-			favoriteSiteIds = Collections.<String>emptyList();
+		Set<String> favoriteSiteIds = Collections.<String>emptySet();
+		List<String> listFavoriteSiteIds = (List<String>)props.getPropertyList(FAVORITES_PROPERTY);
+		if (listFavoriteSiteIds != null) {
+			favoriteSiteIds = new HashSet<String>(props.getPropertyList(FAVORITES_PROPERTY));
```

Comment:
Can this line just be 

favoriteSiteIds = new HashSet<String>(listFavoriteSiteIds);

?

Reply:
Done! that was my initial intention 😅